Description: Support new version of OpenCL.
 This enables support for the new version of OpenCL that ships
 with resolute.
 https://github.com/intel/llvm/issues/20511
Author: Will French <will.french@canonical.com>
Forwarded: not-needed
Last-Update: 2025-11-06
---
--- intel-dpcpp-6.2.0.orig/unified-runtime/source/adapters/opencl/CMakeLists.txt
+++ intel-dpcpp-6.2.0/unified-runtime/source/adapters/opencl/CMakeLists.txt
@@ -57,7 +57,7 @@ if(UR_OPENCL_INCLUDE_DIR)
 else()
     FetchContent_Declare(OpenCL-Headers
         GIT_REPOSITORY  "https://github.com/KhronosGroup/OpenCL-Headers.git"
-        GIT_TAG         542d7a8f65ecfd88b38de35d8b10aa67b36b33b2
+        GIT_TAG         "v2025.07.22"
     )
     FetchContent_MakeAvailable(OpenCL-Headers)
     FetchContent_GetProperties(OpenCL-Headers
@@ -79,7 +79,7 @@ else()
     if(NOT OpenCL_FOUND)
         FetchContent_Declare(OpenCL-ICD-Loader
             GIT_REPOSITORY  "https://github.com/KhronosGroup/OpenCL-ICD-Loader.git"
-            GIT_TAG         main
+            GIT_TAG         "v2025.07.22"
         )
         FetchContent_MakeAvailable(OpenCL-ICD-Loader)
     endif()
@@ -94,6 +94,7 @@ message(STATUS "OpenCL ICD Loader Librar
 target_compile_definitions(ur_adapter_opencl PRIVATE
     CL_TARGET_OPENCL_VERSION=300
     CL_USE_DEPRECATED_OPENCL_1_2_APIS
+    CL_ENABLE_BETA_EXTENSIONS=ON
 )
 
 target_include_directories(${TARGET_NAME} PRIVATE
--- intel-dpcpp-6.2.0.orig/unified-runtime/source/adapters/opencl/memory.cpp
+++ intel-dpcpp-6.2.0/unified-runtime/source/adapters/opencl/memory.cpp
@@ -205,6 +205,8 @@ cl_image_format mapURImageFormatToCL(con
 
 cl_image_desc mapURImageDescToCL(const ur_image_desc_t *PImageDesc) {
   cl_image_desc CLImageDesc;
+  memset(&CLImageDesc, 0, sizeof(CLImageDesc));
+
   CLImageDesc.image_type =
       cl_adapter::cast<cl_mem_object_type>(PImageDesc->type);
 
@@ -237,8 +239,6 @@ cl_image_desc mapURImageDescToCL(const u
   CLImageDesc.image_slice_pitch = PImageDesc->slicePitch;
   CLImageDesc.num_mip_levels = PImageDesc->numMipLevel;
   CLImageDesc.num_samples = PImageDesc->numSamples;
-  CLImageDesc.buffer = nullptr;
-  CLImageDesc.mem_object = nullptr;
 
   return CLImageDesc;
 }
